### ISSUE TITLE
Ability to use a task as a step for another task

### ISSUE THREAD

**Issue by agentmilindu** [leopardslab/Dunner#41]
Ability to  refer `status` task from `deploy` task as `@status`.

```
deploy:
- name: 'golang'
  command: ['go', 'install']
- name: 'mesosphere/aws-cli'
  command: ['aws', 'elasticbeanstalk update-application --application-name myapp']
- name: '@status' #This refers to another task and can pass args too
  args: 'prod'
status:
- name: 'mesosphere/aws-cli'
  command: ['aws', 'elasticbeanstalk describe-events --environment-name $1'] 
```

**Comment by championballer** [comment]
@agentmilindu Can I work on this?

**Comment by agentmilindu** [comment]
This is a bit complex task. take only if you are sure about this. You might need a good understanding of the Dunner code base and Golang.

**Comment by championballer** [comment]
@agentmilindu Is it okay if I can get a few days for this? If I am unable to do by then, then I will turn it over?

**Comment by agentmilindu** [comment]
@championballer How's it going?

**Comment by championballer** [comment]
@agentmilindu Have been reading around this, will go full throttle once exams end.

**Comment by ayushjn20** [comment]
@agentmilindu @championballer 
This feature has a pre-requisite feature described in the issue #3.

**Comment by championballer** [comment]
@ayushjn20 Yeah realised that. Sorry for the delay. Exams ended today. I guess you have a submitted a PR for the same already?

**Comment by agentmilindu** [comment]
Fixed by #47
